DevCycle/Utils/RequestConsolidator.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  RequestConsolidator.swift
+//  DevCycle
+//
+//  Copyright © 2022 Taplytics. All rights reserved.
+//
+
+import Foundation
+
+class RequestConsolidator {
```

Comment:
Instead of rolling alot of this logic ourselves, I wonder if we should be using a built-in iOS mechanism like OperationQueues? 
https://www.raywenderlich.com/books/concurrency-by-tutorials/v2.0/chapters/7-operation-queues
https://developer.apple.com/documentation/foundation/operationqueue

Reply:
we'd still need to implement this logic, operation queues don't solve for this problem https://taplytics.slack.com/archives/C03E0RHMBJ8/p1665675707504839?thread_ts=1665670590.747699&cid=C03E0RHMBJ8